torch/sparse/__init__.py

```diff
@@ -48,12 +48,12 @@ def mm(mat1: Tensor, mat2: Tensor) -> Tensor:
     and dense matrix :attr:`mat2`. Similar to :func:`torch.mm`, If :attr:`mat1` is a
     :math:`(n \times m)` tensor, :attr:`mat2` is a :math:`(m \times p)` tensor, out will be a
     :math:`(n \times p)` dense tensor. :attr:`mat1` need to have `sparse_dim = 2`.
-    This function also supports backward for both matrices. Note that the gradients of
+    This function also supports backward for both matrices. Note that if :attr:`mat1` is a dense tensor, the gradients of
```

Comment:
I'm having a hard time with this existing documentation.

"Performs a matrix multiplication of the sparse matrix :attr:`mat1``: and dense matrix :attr:`mat2`" is how it starts. But then this sentence suggests that :attr:`mat1` may be dense? And the update to args (below) says that :attr:`mat2` might be sparse. 

Can this PR rationalize this paragraph and the args' descriptions?